idxmgmt/manager_test.go

```diff
@@ -157,41 +158,59 @@ func TestManager_Setup(t *testing.T) {
 	var testCasesEnabled = map[string]testCase{
 		"Default": {
 			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeEnabled,
-			tCt: 5, tWithILMCt: 4, pCt: 3, aCt: 3,
+			tCt: 5, tWithILMCt: 4, pCt: 2, aCt: 3,
 		},
 		"LoadTemplateAndILM": {
 			cfg:   common.MapStr{"apm-server.ilm.enabled": true},
 			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeEnabled,
-			tCt: 5, tWithILMCt: 4, pCt: 3, aCt: 3,
+			tCt: 5, tWithILMCt: 4, pCt: 2, aCt: 3,
 		},
 		"DisabledILM": {
 			cfg:   common.MapStr{"apm-server.ilm.enabled": false},
 			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeEnabled,
-			tCt: 5,
+			tCt: 4,
 		},
 		"DisabledTemplate": {
 			cfg:   common.MapStr{"setup.template.enabled": false},
 			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeEnabled,
-			pCt: 3, aCt: 3,
+			pCt: 2, aCt: 3,
 		},
 		"DisabledTemplateAndILM": {
 			cfg:   common.MapStr{"setup.template.enabled": false, "apm-server.ilm.enabled": false},
 			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeEnabled,
 		},
 		"LoadModeDisabledILM": {
 			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeDisabled,
-			tCt: 5, tWithILMCt: 4,
+			tCt: 4, tWithILMCt: 3,
 		},
 		"LoadModeDisabledTemplate": {
 			tLoad: libidxmgmt.LoadModeDisabled, ilmLoad: libidxmgmt.LoadModeEnabled,
-			pCt: 3, aCt: 3,
+			pCt: 2, aCt: 3,
 		},
 		"LoadModeDisabledTemplateAndILM": {
 			tLoad:   libidxmgmt.LoadModeDisabled,
 			ilmLoad: libidxmgmt.LoadModeDisabled,
 		},
 	}
 
+	var testCasesOverwrite = map[string]testCase{
+		"OverwriteILM": {
+			cfg:   common.MapStr{"apm-server.ilm.overwrite": true},
+			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeEnabled,
+			tCt: 5, tWithILMCt: 4, pCt: 4, aCt: 3,
+		},
+		"OverwriteTemplate": {
+			cfg:   common.MapStr{"setup.template.overwrite": true},
+			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeEnabled,
+			tCt: 5, tWithILMCt: 4, pCt: 2, aCt: 3,
+		},
+		"OverwriteTemplateAndILM": {
+			cfg:   common.MapStr{"setup.template.overwrite": true, "apm-server.ilm.overwrite": true},
+			tLoad: libidxmgmt.LoadModeEnabled, ilmLoad: libidxmgmt.LoadModeEnabled,
+			tCt: 5, tWithILMCt: 4, pCt: 4, aCt: 3,
```

Comment:
I really would like to start using abbreviations as an exception rather than as a rule, here and everywhere else. I don't know what is `tCt`, `pCt`, `aCt`, and such. 
Also, this part of the code is quite complicated already, things eg like `supporter.st`, `esIdxCfg` elsewhere don't make it any easier to follow.